src/matching/Generic_vs_generic.ml

```diff
@@ -357,7 +357,12 @@ let m_regexp_options a_opt b_opt =
 (* TODO: factorize with metavariable and aliasing logic in m_expr
  * TODO: remove MV.Id and use always MV.N?
  *)
-let rec m_name a b =
+let rec m_name_inner a b =
+  (* Inside of this function, any recursive call to m_name means to not
+     start another round of entry-specific logic. We only recurse with
+     the inner m_name function.
+  *)
+  let m_name = m_name_inner in
```

Comment:
Why not just change calls to `m_name` with `m_name_inner` ? Seems confusing to have a `let-rec` where `m_name` is defined and within a subset of this `let-rec`, redefine `m_name` to something else.

Reply:
i was afraid of someone who later touched `m_name_inner` forgetting, and accidentally calling the external `m_name`